### ISSUE TITLE
Type 'NexusInputObjectTypeDef<string>' is not assignable to type 'NexustArgDef<string>'.

### ISSUE THREAD

**Issue by marhub** [prisma/nexus#137]
Can't find anything in docs about how to create and register input objects.

What i want to achieve is schema like this:

`type Query{
signup(input: SignupInput): AuthPayload
}`

when i do:

<pre>export const Mutation = mutationType({
  definition(t) {
    t.field('signup', {
      type: 'AuthPayload',
      args: {
        input: inputObjectType({
          name: 'RegisterInput',
          definition: (t)=>{
            t.string('name',{nullable: true }),
              t.string('email'),
              t.string('password')
          }
        })
      },
  (...)</pre>
i have error:
![image](https://user-images.githubusercontent.com/2960432/57570584-da6fe180-7403-11e9-8aea-c1a66b2dbae2.png)

How can i achieve such schema ?

**Comment by stalniy** [comment]
You do it wrong. It needs to be this:

```js
import { arg } from 'nexus'

export const RegisterInput = inputObjectType({
          name: 'RegisterInput',
          definition: (t)=>{
            t.string('name',{nullable: true }),
              t.string('email'),
              t.string('password')
          }
        })

export const Mutation = mutationType({
  definition(t) {
    t.field('signup', {
      type: 'AuthPayload',
      args: {
        input: arg({ type: RegisterInput, required: true })
      },
  (...)
```

**Comment by marhub** [comment]
@stalniy  thanks for your reply, i've already found how to do it - in examples in repository.
I think that should be described in docs :)

**Comment by tgriesser** [comment]
#143 (available on npm under `0.12.0-alpha.1`) allows providing input args without wrapping, and also adds a `asArg()` to input object types, so you can do:

```ts
export const Mutation = mutationType({
  definition(t) {
    t.field('signup', {
      type: 'AuthPayload',
      args: {
        input: RegisterInput.asArg({ required: true })
      },
  (...)
```
